components/kserve/kserve_config_handler.go

```diff
@@ -0,0 +1,148 @@
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+const (
+	KserveConfigMapName string = "inferenceservice-config"
+)
+
+func (k *Kserve) setupKserveConfigAndDependencies(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	// as long as Kserve.Serving is not 'Removed', we will setup the dependencies
+
+	if k.Serving.ManagementState != operatorv1.Removed {
+		if k.DefaultDeploymentMode == Serverless {
+			return fmt.Errorf("setting the defaultDeployment mode Serverless is not compatible with setting the management state for Serving as Removed")
+		}
+		// check on dependent operators if all installed in cluster
+		dependOpsErrors := checkDependentOperators(cli).ErrorOrNil()
+		if dependOpsErrors != nil {
+			return dependOpsErrors
+		}
+
+		if err := k.configureServerless(dscispec); err != nil {
+			return err
+		}
+
+		if err := k.setDefaultDeploymentMode(ctx, cli, dscispec, k.DefaultDeploymentMode); err != nil {
+			return err
+		}
+	}
+
+	// since serverless is specifically set to 'Removed'
+	// we do not need to check if dependent operators are installed and configure them
+	// in this case the default deployment mode is forced to be RawDeployment
+	fmt.Println("Serverless is specified as 'Removed', Kserve will default to RawDeployment")
+	return k.setDefaultDeploymentMode(ctx, cli, dscispec, RawDeployment)
+}
+
+func (k *Kserve) setDefaultDeploymentMode(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec, defaultmode DefaultDeploymentMode) error {
+	inferenceServiceConfigMap := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{
+		Namespace: dscispec.ApplicationsNamespace,
+		Name:      KserveConfigMapName,
+	}, inferenceServiceConfigMap)
+	if err != nil {
+		return fmt.Errorf("error getting configmap 'inferenceservice-config'. %w", err)
+	}
+
+	// set data.deploy.defaultDeploymentMode to the model specified in the Kserve spec
+	var deployData map[string]interface{}
+	if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData); err != nil {
+		return fmt.Errorf("error retrieving value for key 'deploy' from configmap %s. %w", KserveConfigMapName, err)
+	}
+	deployData["defaultDeploymentMode"] = defaultmode
+	deployDataBytes, err := json.MarshalIndent(deployData, "", " ")
+	if err != nil {
+		return fmt.Errorf("could not set values in configmap %s. %w", KserveConfigMapName, err)
+	}
+	inferenceServiceConfigMap.Data["deploy"] = string(deployDataBytes)
+
+	// disabling logic to enable or disable ingress creation till https://github.com/kserve/kserve/pull/3436/files is merged
```

Comment:
Cross reference here: https://github.com/kserve/kserve/pull/3436

Reply:
Thanks @terrytangyuan. How do we want to coordinate this change? 

Should we wait until your PR is merged and we update KServe manifests version pointing to the new release as part of this PR?

Reply:
@bartoszmajsak I forgot to explicitly mention this change in the latest comment. But the idea here is that after @terrytangyuan's PR is merged, we'd rather set the `disableIngressCreation` field according to the deployment mode rather than edit the `ingressClassName`. The logic will not work till the new field from https://github.com/kserve/kserve/pull/3436 is merged into ODH. We can merge this PR independently and have a follow up PR to uncomment this block later, or we can wait for the other change to make it's way in before merging this PR. I'm okay with both

Reply:
@terrytangyuan Do you think KServe would crash if there is a spurious field?
If not, we could commit this code uncommented to avoid the need of a follow-up PR.

Reply:
https://github.com/kserve/kserve/pull/3436 has been merged

Reply:
@VedantMahabaleshwarkar would you mind following-up by uncommenting this code?

Reply:
+1 Otherwise it will leave readers confused seeing this, so that would be great to clean up :)